extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergTable.java

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.util;
+
+import io.deephaven.engine.table.TableDefinition;
+import io.deephaven.engine.table.impl.PartitionAwareSourceTable;
+import io.deephaven.engine.table.impl.SourceTableComponentFactory;
+import io.deephaven.engine.table.impl.locations.TableKey;
+import io.deephaven.engine.updategraph.UpdateSourceRegistrar;
+import io.deephaven.iceberg.layout.IcebergRefreshingTableLocationProvider;
+import io.deephaven.iceberg.location.IcebergTableLocationKey;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+/**
+ * Table wrapper for refreshing Iceberg tables.
+ */
+public class IcebergTable extends PartitionAwareSourceTable {
```

Comment:
I suspect we may want to make `IcebergTable` an interface. Have two implementations, one that throws on `update`, and this one be the other one.